articles/supervised-learning-algorithms/index.md

```diff
@@ -0,0 +1,158 @@
+
+### Supervised Learning
+Let us look at a few of the applications of supervised learning before we dive into the algorithms. Supervised learning tasks require datasets with input-output pairs. Consider the example of trying to classify the digits. Given an image of a digit, what is the number? [MNIST digits](http://yann.lecun.com/exdb/mnist/) dataset is one of the earliest datasets that helped automate the processes of postal services.
+
+Another use case example of supervised learning is predicting the price of houses given a few features. The features can include size, location, facilities, etc. The input consists of the features and the output consists of the price. Algorithms that predict continuous values of data are called regression-based algorithms.
+
+Supervised learning is mainly classified into two types: Classification and Regression. Let us take a closer look at both these algorithm categories.  
+
+#### Classification
+Classification algorithms are a type of supervised learning algorithms that predict outputs from a discrete sample space. For example, predicting a disease, predicting digit output labels such as Yes or No, or 1,2,3, respectively. We can also have scenarios where multiple outputs are required. For this we can consider the example of self-driving cars.
+
+The various objects found on the road need to be classified according to their categories and also need to be classified as safe or unsafe. This scenario is an example of a multi-class classification. We will now look at some of the key algorithms underneath the *classification algorithms*.
+
+1. K-Nearest Neighbors(KNN): KNN is an algorithm that works on creating a decision boundary based on distance metrics. Distance metrics define and parameterize distance. There are various distance metrics such as Euclidean distance, Manhattan distance, etc.
+
+Manhattan distance takes the absolute distance between two points, whereas Euclidean distance takes the square root of the sum of the squares of the distance between the two points.
+
+![distance metrics](/engineering-education/supervised-learning-algorithms/distance.jpg)
+
+*[Image Source](https://www.ieee.ma/uaesb/pdf/distances-in-classification.pdf)*
+
+All machine learning algorithms have hyperparameters to deal with. In K-NN, the parameter is `k`. It is initialized to an integer depending on the number of classes in the dataset known apriori to fitting. `k` signifies the number of nearest points the algorithm considers while creating decision boundaries.  
```

Comment:
"dataset known apriori to fitting." 
This bit was a bit confusing  - just not sure what apriori means - is it a ML term? If so it may be useful to and a link or explain a bit. 🚀

Reply:
apriori means prior to.... I will reword it

Reply:
thank you - I thought that what it may mean - but when I looked it up a ML term also popped up so I was sure. 👍